lib/grpc_reflection/service/builder/util.ex

```diff
@@ -7,13 +7,35 @@ defmodule GrpcReflection.Service.Builder.Util do
 
   @type_message Map.fetch!(Google.Protobuf.FieldDescriptorProto.Type.mapping(), :TYPE_MESSAGE)
 
-  def package_from_name(service_name) do
-    service_name
-    |> String.split(".")
-    |> Enum.reverse()
-    |> then(fn [_ | rest] -> rest end)
-    |> Enum.reverse()
-    |> Enum.join(".")
+  def get_package(module, symbol) do
+    if function_exported?(module, :descriptor, 0) do
+      parent_symbol = symbol |> String.split(".") |> Enum.slice(0..-2) |> Enum.join(".")
+
+      case module.descriptor() do
+        %Google.Protobuf.ServiceDescriptorProto{} -> parent_symbol
+        _ -> handle_parent_module(module, parent_symbol)
+      end
+    else
+      symbol
+    end
```

Comment:
Some of this logic is unnecessary.  Since we know that a package name can't be a service nor a type, we can infer that one of the following is true:
1. The module name that matches the package name will not be an actual module
2. If the package name overlaps with a module name, it shouldn't export `descriptor/0`

Possible alternate implementation:

```elixir
def get_package(symbol) do
  parent_symbol = symbol |> String.split(".") |> Enum.slice(0..-2) |> Enum.join(".")
  parent_module = try_get_parent_module(parent_symbol)
  if function_exported?(parent_module, :descriptor, 0) do
    get_package(parent_symbol)
  else
    parent_symbol
  end
end
```

Reply:
Sure, updated.